src/CommonItilObject_Item.php

```diff
@@ -36,11 +36,1321 @@
+                }
+            }
+        }
+
+        // Show associated item dropdowns
+        if ($canedit) {
+            $p = [
+                'used'       => $params['items_id'],
+                'rand'       => $rand,
+                static::$items_id_1 => $params['id']
+            ];
+        // My items
+            if ($params['_users_id_requester'] > 0) {
+                ob_start();
+                static::dropdownMyDevices($params['_users_id_requester'], $obj->fields["entities_id"], $params['itemtype'], 0, $p);
+                $twig_params['my_items_dropdown'] = ob_get_clean();
+            }
+            // Global search
+            ob_start();
+            static::dropdownAllDevices("itemtype", $params['itemtype'], 0, 1, $params['_users_id_requester'], $obj->fields["entities_id"], $p);
+            $twig_params['all_items_dropdown'] = ob_get_clean();
+
+            //echo "<span id='item_".strtolower(static::$itemtype_1)."_selection_information$rand'></span>";
+            //echo "</div>";
+
+            // Add button
+            //echo "<a href='javascript:itemAction$rand(\"add\");' class='vsubmit' style='float:left'>"._sx('button', 'Add')."</a>";
+        }
+
+        // Display list
+        //echo "<div style='clear:both;'>";
+
+        if (!empty($params['items_id'])) {
+            // No delete if mandatory and only one item
+            $delete = $obj->canAddItem(static::class);
+            $cpt = 0;
+            foreach ($params['items_id'] as $itemtype => $items) {
+                $cpt += count($items);
+            }
+
+            if ($cpt == 1 && isset($tt->mandatory['items_id'])) {
+                $delete = false;
+            }
+            foreach ($params['items_id'] as $itemtype => $items) {
+                foreach ($items as $items_id) {
+                    $count++;
+                    $twig_params['items_to_add'][] = static::showItemToAdd(
+                        $params['id'],
+                        $itemtype,
+                        $items_id,
+                        [
+                            'rand'      => $rand,
+                            'delete'    => $delete,
+                            'visible'   => ($count <= 5)
+                        ]
+                    );
+                }
+            }
+        }
+        $twig_params['count'] = $count;
+        $twig_params['usedcount'] = $usedcount;
+
+        foreach (['id', '_users_id_requester', 'items_id', 'itemtype', '_canupdate'] as $key) {
+            $twig_params['opt'][$key] = $params[$key];
+        }
+
+        TemplateRenderer::getInstance()->display('components/itilobject/add_items.html.twig', $twig_params);
+    }
+
+    /**
+     * Generate JavaScript Code to load the list of itmes
+     *
+     * @param $rand  int random number use for dropdown name
+     * @param $opt   array of possible options to be include in the call
+     *
+     * @return void
+     **/
+    public static function getJs($rand, $opt)
```

Comment:
I'm not 100% sure, but this function seems unused, maybe we can deprecate it (@cedric-anne ?)

Reply:
Thanks, this one was required on 9.5,

the code in version 10 is there https://github.com/glpi-project/glpi/blob/4fb4226b8f8b3559939c1de89eec9bed63a4f41d/templates/components/itilobject/add_items.html.twig#L89

but its hardcoded to 'itemTicket'  I addapted the code